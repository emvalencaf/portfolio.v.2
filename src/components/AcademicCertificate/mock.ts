import { AcademicCertificateProps } from ".";

export default {
	courseTitle: "Curso de React.Js e Next.Js (Intermediário e Avançado)",
	institution: "Udemy, Instrutor: Luiz Otávio Miranda",
	courseWork: "93 horas",
	startIn: 2932983,
	endIn: 3932983,
	urlDownload: "/assets/files/UC-44225967-b7ec-4add-8983-426364ad7bb6.pdf"
} as AcademicCertificateProps;