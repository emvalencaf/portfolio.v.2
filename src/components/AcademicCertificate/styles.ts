import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';

type WrapperAcademicCertificateProps = {
	theme: DefaultTheme;
	showDetails: boolean;
}

const changeCursor = () => css`
	cursor: pointer;
`;

export const Wrapper = styled.div<WrapperAcademicCertificateProps>`
	${({ theme, showDetails }) => css`
	padding: 2rem;
	${!showDetails && changeCursor()}

	& ${HeadingContainer} {
		margin: 0px;
	}
	`}
`;

export const HeaderAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		align-items: center;
		justify-content: center;
		margin: ${theme.spacings.small} auto;
	`}
`;

export const BodyAcademicCertificate = styled.div`
	${({ theme }) => css`
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
		gap: ${theme.spacings.small};
		margin: ${theme.spacings.small} auto;
	`}
`;