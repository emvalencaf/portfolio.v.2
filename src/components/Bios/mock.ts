import { BiosProps } from ".";

export default {
	bios: `
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Inventore rem minus dolore ducimus quos, sed dolores odit maiores quam tempora atque deserunt voluptas perferendis, aspernatur, sequi dignissimos reiciendis. In, esse!
	Lorem ipsum, dolor sit amet consectetur adipisicing elit. Non a deleniti quia. Magnam aspernatur cumque, nemo vero libero excepturi consequuntur corporis ullam nobis molestiae? Atque nam praesentium ullam quaerat quisquam?
</p>
<p>
	Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio cumque recusandae libero sequi, tempora iste, illo, possimus unde cum numquam quae corrupti minima tenetur mollitia earum repellendus animi molestiae voluptates.
	Lorem ipsum dolor sit, amet consectetur adipisicing elit. Obcaecati dolores fugiat ipsam ut odit sed aliquid ducimus molestiae vero ipsa nam veniam dolor libero labore qui pariatur, alias assumenda itaque?
	Lorem ipsum dolor sit amet, consectetur adipisicing elit. Illo cum temporibus nesciunt sequi odit ipsam harum magnam dolore a molestiae. Totam debitis, rerum aliquid sit non ratione. Et, omnis iusto!
</p>
<p>
	Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nam dicta aut cum sequi voluptatibus. Vel pariatur veritatis quidem. Culpa vero deleniti minima earum quod, odit deserunt fugit reprehenderit architecto in!
</p>
	`,
	profilePhoto: {
		altText: "profile picture",
		srcImg: "assets/images/potrait-photo.jpg",
	},
} as BiosProps;
