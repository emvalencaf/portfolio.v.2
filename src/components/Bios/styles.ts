import styled, { css } from "styled-components";
export const Wrapper = styled.div`
	${({ theme }) => css`
		@media ${theme.media.lteMedium} {
			flex-direction: column;
		}

		display: flex;
		flex-direction: row;
		flex-wrap: wrap;
		justify-content: space-between;
		align-items: center;
		gap: ${theme.spacings.xhero};
		text-align: justify;
		& p {
			text-align: justify;
			padding: ${theme.spacings.small};
			max-width: 550px;
		}
	`}
`;

export const ContentContainer = styled.div`
	${() => css`
		max-height: 350px;
		overflow-y: auto;
	`}
`;
