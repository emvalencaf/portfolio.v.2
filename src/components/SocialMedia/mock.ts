import { SocialMediaProps } from ".";

export default {
	instaURL: "https://www.instagram.com/tício/",
	linkedinURL: "https://www.linkedin.com/in/tício/",
	facebookURL: "https://www.facebook.com/tício",
	githubURL: "https://www.github.com/tício",
	twitterURL: "https://www.twitter.com/tício",
	tiktokURL: "https://www.tiktok.com/@tício",
	youtubeURL: "https://www.youtube.com/@tício",
	homepageURL: "https://www.emvalenca.blog.com/",
} as SocialMediaProps;
