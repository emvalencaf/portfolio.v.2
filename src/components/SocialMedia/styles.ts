import styled, { css, DefaultTheme } from 'styled-components';
import { SocialMediaProps } from '.';

const logoSizes = {
	small: (theme: DefaultTheme) => css`
		height: ${theme.font.sizes.medium};
		width: ${theme.font.sizes.medium};
	`,
	medium: (theme: DefaultTheme) => css`
		height: ${theme.font.sizes.large};
		width: ${theme.font.sizes.large};
	`,
	big: (theme: DefaultTheme) => css`
		height: ${theme.font.sizes.xlarge};
		width: ${theme.font.sizes.xlarge};
	`,
};

export const Wrapper = styled.div<SocialMediaProps>`
	${({ theme, sizes }) => css`
		& svg, & a {
			color: ${theme.colors.primaryColor};
			${logoSizes[sizes](theme)}
			transition: all 350ms ease-in-out;
		}

		&  svg:hover {
			transform: scale(1.2);
			opacity: 0.8;
		}
	`}
`;

export const SocialMediaLink = styled.a`
	${( {theme} ) => css`
		color: ${theme.colors.primaryColor};
	`}
`;
