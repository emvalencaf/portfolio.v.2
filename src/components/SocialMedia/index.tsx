// icons
import {
	FacebookCircle,
	Github,
	InstagramAlt,
	Linkedin,
	Tiktok,
	Twitter,
	Youtube,
} from "@styled-icons/boxicons-logos";
import { PersonPin } from "@styled-icons/material-outlined";
// styles
import * as Styled from "./styles";

// types
export type SocialMediaProps = {
	instaURL?: string;
	linkedinURL?: string;
	facebookURL?: string;
	homepageURL?: string;
	twitterURL?: string;
	githubURL?: string;
	youtubeURL?: string;
	tiktokURL?: string;
	sizes?: "small" | "medium" | "big";
};

const urlPattern = {
	github: "github.com/",
	linkedin: "linkedin.com/in/",
	instagram: "instagram.com/",
	twitter: "twitter.com/",
	facebook: "facebook.com/",
	youtube: "youtube.com/",
	tiktok: "tiktok.com/",
};

const SocialMedia = ({
	instaURL = "",
	linkedinURL = "",
	facebookURL = "",
	homepageURL = "",
	twitterURL = "",
	githubURL = "",
	youtubeURL = "",
	tiktokURL = "",
	sizes = "medium",
}: SocialMediaProps) => {
	return (
		<Styled.Wrapper sizes={sizes}>
			{facebookURL && facebookURL.includes(urlPattern.facebook) ? (
				<Styled.SocialMediaLink
					rel="external"
					target="_blank"
					href={facebookURL}
					aria-label="facebook profile"
				>
					<FacebookCircle />
				</Styled.SocialMediaLink>
			) : (
				""
			)}
			{instaURL && instaURL.includes(urlPattern.instagram) ? (
				<Styled.SocialMediaLink
					rel="external"
					target="_blank"
					href={instaURL}
					aria-label="instagram profile"
				>
					<InstagramAlt />
				</Styled.SocialMediaLink>
			) : (
				""
			)}
			{linkedinURL && linkedinURL.includes(urlPattern.linkedin) ? (
				<Styled.SocialMediaLink
					rel="external"
					target="_blank"
					href={linkedinURL}
					aria-label="linkedin profile"
				>
					<Linkedin />
				</Styled.SocialMediaLink>
			) : (
				""
			)}
			{githubURL && githubURL.includes(urlPattern.github) ? (
				<Styled.SocialMediaLink
					rel="external"
					target="_blank"
					href={githubURL}
					aria-label="github profile"
				>
					<Github />
				</Styled.SocialMediaLink>
			) : (
				""
			)}
			{twitterURL && twitterURL.includes(urlPattern.twitter) ? (
				<Styled.SocialMediaLink
					rel="external"
					target="_blank"
					href={twitterURL}
					aria-label="twitter profile"
				>
					<Twitter />
				</Styled.SocialMediaLink>
			) : (
				""
			)}
			{youtubeURL && youtubeURL.includes(urlPattern.youtube) ? (
				<Styled.SocialMediaLink
					rel="external"
					target="_blank"
					href={youtubeURL}
					aria-label="youtube profile"
				>
					<Youtube />
				</Styled.SocialMediaLink>
			) : (
				""
			)}
			{tiktokURL && tiktokURL.includes(urlPattern.tiktok) ? (
				<Styled.SocialMediaLink
					rel="external"
					target="_blank"
					href={instaURL}
					aria-label="tiktok profile"
				>
					<Tiktok />
				</Styled.SocialMediaLink>
			) : (
				""
			)}
			{homepageURL ? (
				<Styled.SocialMediaLink
					rel="external"
					target="_blank"
					href={homepageURL}
					aria-label="personal blog"
				>
					<PersonPin />
				</Styled.SocialMediaLink>
			) : (
				""
			)}
		</Styled.Wrapper>
	);
};

export default SocialMedia;
