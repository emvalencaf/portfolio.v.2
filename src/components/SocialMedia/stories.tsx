import { Meta, Story } from "@storybook/react/types-6-0";
import SocialMedia, { SocialMediaProps } from ".";
import mock from "./mock";

export default {
	title: "SocialMedia",
	component: SocialMedia,
	args: mock,
	argTypes: {
		instaURL: { type: "string" },
		facebookURL: { type: "string" },
		twitterURL: { type: "string" },
		githubURL: { type: "string" },
		homepageURL: { type: "string" },
		linkedinURL: { type: "string" },
	},
} as Meta<SocialMediaProps>;

export const Template: Story<SocialMediaProps> = (args) => {
	return (
		<div>
			<SocialMedia {...args} />
		</div>
	);
};
