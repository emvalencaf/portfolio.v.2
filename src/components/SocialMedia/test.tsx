import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";

// theme
import { theme } from "../../styles/theme";

// react component to be test
import SocialMedia from ".";

// mock
import mock from "./mock";

describe("<SocialMedia />", () => {
	it("should render all social media in SocialMedia component", () => {
		renderTheme(<SocialMedia {...mock} />);

		expect(screen.getAllByRole("link")).toHaveLength(8);
	});

	it("should rend no social media in SocialMedia component", () => {
		renderTheme(<SocialMedia />);

		expect(screen.queryAllByRole("link")).toHaveLength(0);
	});

	it("should rend SocialMedia component with sizes small", () => {
		renderTheme(<SocialMedia githubURL={mock.githubURL} sizes="small" />);

		expect(screen.getByLabelText("github profile")).toHaveStyle({
			width: theme.fonts.sizes.medium,
			height: theme.fonts.sizes.medium,
		});
	});

	it("should rend SocialMedia component with size medium", () => {
		renderTheme(<SocialMedia githubURL={mock.githubURL} sizes="medium" />);
		expect(screen.getByLabelText("github profile")).toHaveStyle({
			width: theme.fonts.sizes.large,
			height: theme.fonts.sizes.large,
		});
	});

	it("should rend SocialMedia component with size big", () => {
		renderTheme(<SocialMedia githubURL={mock.githubURL} sizes="big" />);
		expect(screen.getByLabelText("github profile")).toHaveStyle({
			width: theme.fonts.sizes.xlarge,
			height: theme.fonts.sizes.xlarge,
		});
	});

	it("should match a snapshot", () => {
		const { container } = renderTheme(<SocialMedia {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
