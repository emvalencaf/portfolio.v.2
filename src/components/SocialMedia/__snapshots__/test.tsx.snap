// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialMedia /> should match a snapshot 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 svg,
.c0 a {
  color: #0A1128;
  height: 3.2rem;
  width: 3.2rem;
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
}

.c0 svg:hover {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
  opacity: 0.8;
}

.c1 {
  color: #0A1128;
}

<div>
  <div
    class="c0"
    sizes="medium"
  >
    <a
      aria-label="facebook profile"
      class="c1"
      href="https://www.facebook.com/tício"
      rel="external"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.001 2.002c-5.522 0-9.999 4.477-9.999 9.999 0 4.99 3.656 9.126 8.437 9.879v-6.988h-2.54v-2.891h2.54V9.798c0-2.508 1.493-3.891 3.776-3.891 1.094 0 2.24.195 2.24.195v2.459h-1.264c-1.24 0-1.628.772-1.628 1.563v1.875h2.771l-.443 2.891h-2.328v6.988C18.344 21.129 22 16.992 22 12.001c0-5.522-4.477-9.999-9.999-9.999z"
        />
      </svg>
    </a>
    <a
      aria-label="instagram profile"
      class="c1"
      href="https://www.instagram.com/tício/"
      rel="external"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.947 8.305a6.53 6.53 0 0 0-.419-2.216 4.61 4.61 0 0 0-2.633-2.633 6.606 6.606 0 0 0-2.186-.42c-.962-.043-1.267-.055-3.709-.055s-2.755 0-3.71.055a6.606 6.606 0 0 0-2.185.42 4.607 4.607 0 0 0-2.633 2.633 6.554 6.554 0 0 0-.419 2.185c-.043.963-.056 1.268-.056 3.71s0 2.754.056 3.71c.015.748.156 1.486.419 2.187a4.61 4.61 0 0 0 2.634 2.632 6.584 6.584 0 0 0 2.185.45c.963.043 1.268.056 3.71.056s2.755 0 3.71-.056a6.59 6.59 0 0 0 2.186-.419 4.615 4.615 0 0 0 2.633-2.633c.263-.7.404-1.438.419-2.187.043-.962.056-1.267.056-3.71-.002-2.442-.002-2.752-.058-3.709zm-8.953 8.297c-2.554 0-4.623-2.069-4.623-4.623s2.069-4.623 4.623-4.623a4.623 4.623 0 0 1 0 9.246zm4.807-8.339a1.077 1.077 0 0 1-1.078-1.078 1.077 1.077 0 1 1 2.155 0c0 .596-.482 1.078-1.077 1.078z"
        />
        <circle
          cx="11.994"
          cy="11.979"
          r="3.003"
        />
      </svg>
    </a>
    <a
      aria-label="linkedin profile"
      class="c1"
      href="https://www.linkedin.com/in/tício/"
      rel="external"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="4.983"
          cy="5.009"
          r="2.188"
        />
        <path
          d="M9.237 8.855v12.139h3.769v-6.003c0-1.584.298-3.118 2.262-3.118 1.937 0 1.961 1.811 1.961 3.218v5.904H21v-6.657c0-3.27-.704-5.783-4.526-5.783-1.835 0-3.065 1.007-3.568 1.96h-.051v-1.66H9.237zm-6.142 0H6.87v12.139H3.095z"
        />
      </svg>
    </a>
    <a
      aria-label="github profile"
      class="c1"
      href="https://www.github.com/tício"
      rel="external"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M12.026 2c-5.509 0-9.974 4.465-9.974 9.974 0 4.406 2.857 8.145 6.821 9.465.499.09.679-.217.679-.481 0-.237-.008-.865-.011-1.696-2.775.602-3.361-1.338-3.361-1.338-.452-1.152-1.107-1.459-1.107-1.459-.905-.619.069-.605.069-.605 1.002.07 1.527 1.028 1.527 1.028.89 1.524 2.336 1.084 2.902.829.091-.645.351-1.085.635-1.334-2.214-.251-4.542-1.107-4.542-4.93 0-1.087.389-1.979 1.024-2.675-.101-.253-.446-1.268.099-2.64 0 0 .837-.269 2.742 1.021a9.582 9.582 0 0 1 2.496-.336 9.554 9.554 0 0 1 2.496.336c1.906-1.291 2.742-1.021 2.742-1.021.545 1.372.203 2.387.099 2.64.64.696 1.024 1.587 1.024 2.675 0 3.833-2.33 4.675-4.552 4.922.355.308.675.916.675 1.846 0 1.334-.012 2.41-.012 2.737 0 .267.178.577.687.479C19.146 20.115 22 16.379 22 11.974 22 6.465 17.535 2 12.026 2z"
          fill-rule="evenodd"
        />
      </svg>
    </a>
    <a
      aria-label="twitter profile"
      class="c1"
      href="https://www.twitter.com/tício"
      rel="external"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.633 7.997c.013.175.013.349.013.523 0 5.325-4.053 11.461-11.46 11.461-2.282 0-4.402-.661-6.186-1.809.324.037.636.05.973.05a8.07 8.07 0 0 0 5.001-1.721 4.036 4.036 0 0 1-3.767-2.793c.249.037.499.062.761.062.361 0 .724-.05 1.061-.137a4.027 4.027 0 0 1-3.23-3.953v-.05c.537.299 1.16.486 1.82.511a4.022 4.022 0 0 1-1.796-3.354c0-.748.199-1.434.548-2.032a11.457 11.457 0 0 0 8.306 4.215c-.062-.3-.1-.611-.1-.923a4.026 4.026 0 0 1 4.028-4.028c1.16 0 2.207.486 2.943 1.272a7.957 7.957 0 0 0 2.556-.973 4.02 4.02 0 0 1-1.771 2.22 8.073 8.073 0 0 0 2.319-.624 8.645 8.645 0 0 1-2.019 2.083z"
        />
      </svg>
    </a>
    <a
      aria-label="youtube profile"
      class="c1"
      href="https://www.youtube.com/@tício"
      rel="external"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21.593 7.203a2.506 2.506 0 0 0-1.762-1.766C18.265 5.007 12 5 12 5s-6.264-.007-7.831.404a2.56 2.56 0 0 0-1.766 1.778c-.413 1.566-.417 4.814-.417 4.814s-.004 3.264.406 4.814c.23.857.905 1.534 1.763 1.765 1.582.43 7.83.437 7.83.437s6.265.007 7.831-.403a2.515 2.515 0 0 0 1.767-1.763c.414-1.565.417-4.812.417-4.812s.02-3.265-.407-4.831zM9.996 15.005l.005-6 5.207 3.005-5.212 2.995z"
        />
      </svg>
    </a>
    <a
      aria-label="tiktok profile"
      class="c1"
      href="https://www.instagram.com/tício/"
      rel="external"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.59 6.69a4.83 4.83 0 0 1-3.77-4.25V2h-3.45v13.67a2.89 2.89 0 0 1-5.2 1.74 2.89 2.89 0 0 1 2.31-4.64 2.93 2.93 0 0 1 .88.13V9.4a6.84 6.84 0 0 0-1-.05A6.33 6.33 0 0 0 5 20.1a6.34 6.34 0 0 0 10.86-4.43v-7a8.16 8.16 0 0 0 4.77 1.52v-3.4a4.85 4.85 0 0 1-1-.1z"
        />
      </svg>
    </a>
    <a
      aria-label="personal blog"
      class="c1"
      href="https://www.emvalenca.blog.com/"
      rel="external"
      target="_blank"
    >
      <svg
        aria-hidden="true"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 2H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h4l3 3 3-3h4c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 16h-4.83l-.59.59L12 20.17l-1.59-1.59-.58-.58H5V4h14v14zm-7-7c1.65 0 3-1.35 3-3s-1.35-3-3-3-3 1.35-3 3 1.35 3 3 3zm0-4c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm6 8.58c0-2.5-3.97-3.58-6-3.58s-6 1.08-6 3.58V17h12v-1.42zM8.48 15c.74-.51 2.23-1 3.52-1s2.78.49 3.52 1H8.48z"
        />
      </svg>
    </a>
  </div>
</div>
`;
